common/src/test/java/io/druid/common/guava/CombiningSequenceTest.java

```diff
@@ -271,4 +274,90 @@ public boolean apply(
     Assert.assertFalse(expectedVals.hasNext());
     yielder.close();
   }
+
+  @Test
+  public void testFails()
+  {
+    List<Integer> output = new ArrayList<>();
+
+    Sequence<Integer> weirdSequence = getWeirdSequence();
+
+    Yielder<Integer> yielder = weirdSequence.toYielder(
+        null,
+        new YieldingAccumulator<Integer, Integer>()
+        {
+          @Override
+          public Integer accumulate(Integer accumulated, Integer in)
+          {
+            yield();
+            return in;
+          }
+        }
+    );
+
+    while (!yielder.isDone()) {
+      output.add(yielder.get());
+      yielder = yielder.next(null);
+    }
+
+    Assert.assertEquals(8, Iterables.getOnlyElement(output).intValue());
+  }
+
+  @Test
+  public void testPasses()
```

Comment:
@himanshug can you help @navis clean up the tests so they make sense for someone reading the code out of context in the future?